Tests/XcodeGenKitTests/SourceGeneratorTests.swift

```diff
@@ -517,6 +488,141 @@ class SourceGeneratorTests: XCTestCase {
+                    let project = Project(basePath: directoryPath, name: "Test", targets: [target])
+
+                    let pbxProj = try project.generatePbxProj()
+
+                    for ex in expectations {
+                        do {
+                            try pbxProj.expectFile(paths: ["Sources", "Hello.\(ex)", "a.swift"], buildPhase: .sources)
+                        } catch {
+                            // e.g. Hello.sgi becomes Hello.SGI.
+                            try pbxProj.expectFile(paths: ["Sources", "Hello.\(ex.uppercased())", "a.swift"], buildPhase: .sources)
+                        }
+
+                        try pbxProj.expectFileMissing(paths: ["Sources", "Hello.\(ex)"])
+                    }
+                }
+            }
+
+            $0.context("when they have utis other than public.folder or dyn* utis") {
+
+                $0.it("treats as resources") {
+
+                    var directories = ["""
+                        Sources:
+                        """]
+
+                    for ex in allSpecialFileExtensions {
+                        directories.append("""
+                          Hello.\(ex):
+                          - a.swift
+                        """)
+                    }
+
+                    try createDirectories(directories.joined(separator: "\n"))
+
+                    let target = Target(name: "Test", type: .application, platform: .iOS, sources: ["Sources"])
+                    let project = Project(basePath: directoryPath, name: "Test", targets: [target])
+
+                    let pbxProj = try project.generatePbxProj()
+                    var errors: [Error] = []
+
+                    // These are ignored in this test.
+                    // They are handled differently and end-up with other BulidPhases.
+                    let ignore = ["xcdatamodeld", "framework", "xpc", "lproj"]
+
+                    for ex in allSpecialFileExtensions.filter({ !ignore.contains($0) }) {
+                        do {
+                            try pbxProj.expectFile(paths: ["Sources", "Hello.\(ex)"], buildPhase: .resources)
+                        } catch {
+                            errors.append(error)
+                        }
+
+                        do {
+                            try pbxProj.expectFileMissing(paths: ["Sources", "Hello.\(ex)", "a.swift"])
+                        } catch {
+                            errors.append(error)
+                        }
+                    }
+
+                    if !errors.isEmpty {
+                        throw failure(errors.map { "\($0)" }.joined(separator: "\n"))
+                    }
+
+                    let groupPath = pbxProj.objects.versionGroups.values.first?.path
+                    try expect(groupPath) == "Hello.xcdatamodeld"
+                }
+            }
+        }
+    }
+}
+
+// MARK: Utilities
```

Comment:
Moved utility functions below the all test cases. What we are interested in most is the test cases themselves, not utilities.